universal_computation/fpt.py

```diff
@@ -1,5 +1,75 @@
-import torch
-import torch.nn as nn
+from transformers import GPT2Model
+
+from .ViT import *
+
+
+def get_nlp_transformer(model_name: str, pretrained: bool):
+    pretrained_transformer = GPT2Model.from_pretrained(model_name)
+    if pretrained:
+        transformer = pretrained_transformer
+    else:
+        transformer = GPT2Model(pretrained_transformer.config)
+
+    return transformer, transformer.config.n_embd
+
+
+class ViTEmbeddings(nn.Module):
+    def __init__(self, config, max_sequence_len: int):
+        super().__init__()
+        self.position_embeddings = nn.Parameter(torch.zeros(1, max_sequence_len + 1, config.hidden_size))
+        self.cls_token = nn.Parameter(torch.zeros(1, 1, config.hidden_size))
+
+    def forward(self, embeddings):
+        bs = embeddings.shape[0]
+        seq_len = embeddings.shape[1]
+        cls_token = self.cls_token.expand(bs, -1, -1)
+        embeddings = torch.cat((cls_token, embeddings), dim=1)
+        embeddings = embeddings + self.position_embeddings[:, :seq_len + 1, :]
+        return embeddings
+
+
+class Lambda(nn.Module):
+    def forward(self, output):
+        x = output[0]
+        return x[:, 0]
+
+
+def get_vit_transformer(model_name: str, pretrained: bool, max_sequence_len: int = 1024):
+    config = CONFIGS[model_name]
+    model = VisionTransformer(config, num_classes=1000, img_size=384)
+    if pretrained:
+        model.load_from(np.load(f'{model_name}.npz'))
+    transformer = nn.Sequential(ViTEmbeddings(config, max_sequence_len), model.transformer.encoder, Lambda())
+    print(transformer)
```

Comment:
Is it a good practice to print(transformer) every time?

Reply:
Hell no!